packages/utils/telemetry-utils/src/telemetryTypes.ts

```diff
@@ -0,0 +1,90 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryBaseLogger, TelemetryEventCategory } from "@fluidframework/common-definitions";
+
+/**
+ * Property types that can be logged.
+ */
+export type TelemetryEventPropertyTypeExt =
+    | string
+    | number
+    | boolean
+    | undefined
+    | (string | number | boolean)[];
+
+ /**
+  * A property to be logged to telemetry containing both the value and a tag. Tags are generic strings that can be used
+  * to mark pieces of information that should be organized or handled differently by loggers in various first or third
+  * party scenarios. For example, tags are used to mark personal information that should not be stored in logs.
+  */
+export interface ITaggedTelemetryPropertyTypeExt {
+    value: TelemetryEventPropertyTypeExt;
+    tag: string;
+}
+
+/**
+ * JSON-serializable properties, which will be logged with telemetry.
+ */
+export interface ITelemetryPropertiesExt {
+    [index: string]: TelemetryEventPropertyTypeExt | ITaggedTelemetryPropertyTypeExt;
+}
+
+/**
+ * Base interface for logging telemetry statements.
+ * Can contain any number of properties that get serialized as json payload.
+ * @param category - category of the event, like "error", "performance", "generic", etc.
+ * @param eventName - name of the event.
+ */
+ export interface ITelemetryEventExt extends ITelemetryPropertiesExt {
+    category: string;
+    eventName: string;
+}
+
+/**
+ * Informational (non-error) telemetry event
+ * Maps to category = "generic"
+ */
+ export interface ITelemetryGenericEventExt extends ITelemetryPropertiesExt {
+    eventName: string;
+    category?: TelemetryEventCategory;
+}
+
+/**
+ * Error telemetry event.
+ * Maps to category = "error"
+ */
+export interface ITelemetryErrorEventExt extends ITelemetryPropertiesExt {
+    eventName: string;
+}
+
+/**
+ * Performance telemetry event.
+ * Maps to category = "performance"
+ */
+export interface ITelemetryPerformanceEventExt extends ITelemetryGenericEventExt {
+    duration?: number; // Duration of event (optional)
+}
+
+export interface ITelemetryLoggerExt extends ITelemetryBaseLogger {
```

Comment:
Comment please.  Maybe mention that this is for use internally within FF code, and ITelemetryBaseLogger should be used when loggers are passed between layers.